src/perform_step/firk_perform_step.jl

```diff
@@ -0,0 +1,158 @@
+function initialize!(integrator, cache::RadauIIA5ConstantCache)
+  integrator.kshortsize = 2
+  integrator.k = typeof(integrator.k)(undef, integrator.kshortsize)
+  integrator.fsalfirst = integrator.f(integrator.uprev, integrator.p, integrator.t) # Pre-start fsal
+
+  # Avoid undefined entries if k is an array of arrays
+  integrator.fsallast = zero(integrator.fsalfirst)
+  integrator.k[1] = integrator.fsalfirst
+  integrator.k[2] = integrator.fsallast
+end
+
+@muladd function perform_step!(integrator, cache::RadauIIA5ConstantCache, repeat_step=false)
+  @unpack t,dt,uprev,u,f,p = integrator
+  @unpack T11, T12, T13, T21, T22, T23, T31, TI11, TI12, TI13, TI21, TI22, TI23, TI31, TI32, TI33 = cache.tab
+  @unpack c1, c2, γ, α, β, e1, e2, e3 = cache.tab
+  @unpack tol, κ, cont1, cont2, cont3 = cache
+  @unpack internalnorm, abstol, reltol, adaptive = integrator.opts
+  alg = unwrap_alg(integrator, true)
+  @unpack min_iter, max_iter = alg
+  mass_matrix = integrator.f.mass_matrix
+  is_compos = integrator.alg isa CompositeAlgorithm
+
+  # precalculations
+  c1m1 = c1-1
+  c2m1 = c2-1
+  c1mc2= c1-c2
+  κtol = κ*tol # used in Newton iteration
+  γdt, αdt, βdt = γ/dt, α/dt, β/dt
+  J = calc_J(integrator, cache, is_compos)
+  if u isa Number
+    LU1 = γdt*mass_matrix - J
```

Comment:
this term should be mass_matrix/(gamma*dt), but it's not

Reply:
nevermind

Reply:
wait, yes it is.

Reply:
<img width="343" alt="capture" src="https://user-images.githubusercontent.com/1814174/51450158-837fb880-1cfd-11e9-9c9a-441ba79eff68.PNG">

Reply:
In all cases you should have `gamma*dt`. The transformation is to divide both sides by `gdt = gamma*dt`. The reason is because `J` is O(n^2) but `I` is `O(n)` (without mass matrices of course), so this decreases the overall calculations. This is exactly the same as the `invW_t` transformation on the Rosenbrock and SDIRK methods.